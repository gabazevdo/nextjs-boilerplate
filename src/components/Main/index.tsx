import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="imagem de um átomo, simbolizando o Logotipo do Curso e ao lado escrito React Avançado"
    />
    <S.Title>React Avançado</S.Title>
    <S.Descrioption>
      Typescript, ReactJS, NExtJS and Styled Components
    </S.Descrioption>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="imagem de um desenvolvedor de frente para o computador"
    />
  </S.Wrapper>
)

export default Main
