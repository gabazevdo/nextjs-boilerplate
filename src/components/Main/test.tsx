import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main/>', () => {
  it('should render the reheading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /Boilerplate NextJS/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should contain background color', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b'
    })
  })
})
