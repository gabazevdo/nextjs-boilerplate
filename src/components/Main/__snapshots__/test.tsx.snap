// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render the reheading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1.3rem;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="imagem de um átomo, simbolizando o Logotipo do Curso e ao lado escrito React Avançado"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    Boilerplate NextJS
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS, Styled Components and Jest for Testing
  </h2>
  <img
    alt="imagem de um desenvolvedor de frente para o computador"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
